[Bug] v10.5.2: null is not an object (getChromaSync().syncUserPrompt) when CLAUDE_MEM_CHROMA_ENABLED=false

## Bug Description

After setting `CLAUDE_MEM_CHROMA_ENABLED=false` in `~/.claude-mem/settings.json` to disable Chroma (as recommended in #707), the worker-service throws a null pointer error on every `UserPromptSubmit` / session init:

```
[ERROR] [HTTP  ] Route handler error {path=/sessions/48/init} null is not an object (evaluating 'this.dbManager.getChromaSync().syncUserPrompt')
[ERROR] [WORKER] ✗ Request failed null is not an object (evaluating 'this.dbManager.getChromaSync().syncUserPrompt')
[ERROR] [HOOK  ] ✗ SDK agent start failed: 500 {sessionDbId=48, promptNumber=1}
```

The worker correctly logs `Chroma disabled via CLAUDE_MEM_CHROMA_ENABLED=false, using SQLite-only search` on startup, but call sites that call `this.dbManager.getChromaSync().syncUserPrompt(...)` (and similar methods) don't guard against `null`.

## Root Cause

`getChromaSync()` returns `null` when Chroma is disabled. The code calls methods on it without a null check. There are approximately **18 occurrences** of `getChromaSync` / `syncUserPrompt` / `chromaSync` in `worker-service.cjs` that are potentially unguarded.

## Environment

- **claude-mem version:** 10.5.2  
- **Platform:** macOS (Apple M1 Max, macOS 25.3.0)  
- **Setting:** `CLAUDE_MEM_CHROMA_ENABLED=false` in `~/.claude-mem/settings.json`  
- **Claude Code version:** 2.1.70

## Steps to Reproduce

1. Set `"CLAUDE_MEM_CHROMA_ENABLED": "false"` in `~/.claude-mem/settings.json`
2. Kill any running chroma-mcp processes
3. Restart the worker-service (or start a new Claude Code session)
4. On first `UserPromptSubmit`, observe the 500 error in `~/.claude-mem/logs/claude-mem-YYYY-MM-DD.log`

## Expected Behavior

When `CLAUDE_MEM_CHROMA_ENABLED=false`, all `getChromaSync()` call sites should guard against `null` (e.g. `getChromaSync()?.syncUserPrompt(...)`). Session init should succeed and fall back to SQLite-only operation.

## Impact

- Session observations may not be saved for the affected session
- The `SessionStart` hook shows an error in the Claude Code UI
- System remains stable (no crash), but memory tracking is partially broken

## Context

This was discovered after disabling Chroma to fix an uncontrolled disk growth issue where `~/.claude-mem/chroma/link_lists.bin` grew to ~420GB as a sparse file on APFS, crashing the system. The SQLite-only mode (#707) is exactly the right fix for that scenario, but this null-pointer regression prevents it from working cleanly.

--- a/src/services/worker/SearchManager.ts
+++ b/src/services/worker/SearchManager.ts
@@ -61,6 +61,9 @@ export class SearchManager {
     limit: number,
     whereFilter?: Record<string, any>
   ): Promise<{ ids: number[]; distances: number[]; metadatas: any[] }> {
+    if (!this.chromaSync) {
+      return { ids: [], distances: [], metadatas: [] };
+    }
     return await this.chromaSync.queryChroma(query, limit, whereFilter);
   }
 

--- a/src/services/worker/http/routes/SessionRoutes.ts
+++ b/src/services/worker/http/routes/SessionRoutes.ts
@@ -356,7 +356,7 @@ export class SessionRoutes extends BaseRouteHandler {
       // Sync user prompt to Chroma
       const chromaStart = Date.now();
       const promptText = latestPrompt.prompt_text;
-      this.dbManager.getChromaSync().syncUserPrompt(
+      this.dbManager.getChromaSync()?.syncUserPrompt(
         latestPrompt.id,
         latestPrompt.memory_session_id,
         latestPrompt.project,

